dvc/repo/__init__.py

```diff
@@ -78,7 +79,11 @@ def __init__(self, root_dir=None):
 
         self.config = Config(self.dvc_dir)
 
-        self.scm = SCM(self.root_dir)
+        no_scm = self.config["core"].get("no_scm", False)
+        if no_scm:
+            self.scm = NoSCM()
+        else:
+            self.scm = SCM(root_dir)
```

Comment:
So if there is no `no-scm=True` and no git, `SCM` will fallback to `NoSCM` silently, which doesnt look right.

Reply:
Fixed, SCM is handling this now. If no_scm is True and there is no git, we will get an exception.